### PR TITLE
[Merged by Bors] - fix: get rid of AWK and its implementation differences from `start_port.sh`

### PR DESCRIPTION
**Pull request by Parcly-Taxel**
This PR replaces the AWK script used in `start_port.sh` – introduced in #1523 – with a Python equivalent that is insensitive to any implementation difference.

By not having to worry about implementations I believe this is a superior solution to the problem that #2794 tackles.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: scripts/start_port.sh [hunk 1]
```diff
@@ -67,8 +67,24 @@ echo "Applying automated fixes"
...
+        lns.append(ln)
+        if len(lns) > 1 and lns[1] == "  by" and len(lns[0]) < 98 and not lns[0].lstrip().startswith("--"):
+            lns.pop()
+            lns[0] += " by"
+        elif len(lns) > 1:
+            print(lns[0], file=g)
+    lns.popleft()
+    for ln in lns:
+        print(ln, file=g)
+BYBACK
```
Comment (eric-wieser) [inline scripts/start_port.sh:86]: Can you move this to `fix-line-breaks.py` so that it's next to `fix-line-breaks.sh`?

Comment (Parcly-Taxel) [reply scripts/start_port.sh]: Done, though I am not entirely sure whether the method of getting the new Python file's path is fully cross-platform.

File: scripts/start_port.sh [hunk 2]
```diff
@@ -67,8 +68,7 @@ echo "Applying automated fixes"
     sed -i 's/Mathbin\./Mathlib\./g' "$mathlib4_path"
     sed -i '/^import/{s/[.]Gcd/.GCD/g; s/[.]Modeq/.ModEq/g; s/[.]Nary/.NAry/g; s/[.]Peq/.PEq/g; s/[.]Pfun/.PFun/g; s/[.]Pnat/.PNat/g; s/[.]Smul/.SMul/g; s/[.]Zmod/.ZMod/g}' "$mathlib4_path"
 
-    # awk script taken from https://github.com/leanprover-community/mathlib4/pull/1523
-    awk '{do {{if (match($0, "^  by$") && length(p) < 98 && (!(match(p, "^[ \t]*--.*$")))) {p=p " by";} else {if (NR!=1) {print p}; p=$0}}} while (getline == 1) if (getline==0) print p}' "$mathlib4_path" > "$mathlib4_path.tmp"
+    python3 $root_path/scripts/fix-line-breaks.py "$mathlib4_path" "$mathlib4_path.tmp"
```
Comment (eric-wieser) [inline scripts/start_port.sh:71]: Could also use 
```suggestion
    python3 "$(dirname "${BASH_SOURCE[0]}")"/fix-line-breaks.py "$mathlib4_path" "$mathlib4_path.tmp"
```
(untested)

Comment (Parcly-Taxel) [reply scripts/start_port.sh:71]: That doesn't work on my system as-is – at that point in the shell script the changed part returns the temporary directory created for the just-ported file.

**Review by eric-wieser — APPROVED** [review]
LGTM, thanks

Ok with you, @jcommelin?

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**